content/en/blog/2023/any-metric-receiver.md

```diff
@@ -0,0 +1,480 @@
+Timestamp: 2023-11-24 13:49:03 +0000 UTC
+Value: 4349619
+	{"kind": "exporter", "data_type": "metrics", "name": "debug"}
+```
+
+The Transform Processor and OTTL offer a wide range of capabilities. You can
+learn more by reviewing the
+[OTTL README.md](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/README.md)
+and watching the talk
+[OTTL Me Why Transforming Telemetry in the OpenTelemetry Collector Just Got Better](https://www.youtube.com/watch?v=uVs0oUV72CE)
+by [Tyler Helmuth](https://github.com/TylerHelmuth) and
+[Evan Bradley](https://github.com/evan-bradley). With this you are ready to
+receive any custom metric with the OpenTelemetry Collector!
+
+## Bonus: Use OTLP!
+
+While the use of the Carbon Receiver and the Transform Processor is a dependable
+method to gather custom metrics, it may seem unconventional to use an external
+format to import metrics into OpenTelemetry, especially when the
+[OpenTelemetry Protocol](/docs/specs/otlp/) (OTLP) provides everything you need!
+
+So, as an alternative to using the Carbon Receiver, you can also transmit a
+custom metric using
+[OTLP JSON](https://github.com/open-telemetry/opentelemetry-proto/tree/main/examples):
+
+```shell
+#!/bin/bash
+URL=${1}
+PORT=${2:-443}
+
+now=$(date +%s)
+notAfterString=$(echo q | openssl s_client -servername "${URL}" "${URL}:${PORT}" 2>/dev/null | openssl x509 -noout -enddate | awk -F"=" '{ print $2; }')
+notAfter=$(date -d "${notAfterString}" +%s)
+
+secondsLeft=$(($notAfter-$now))
+
+data="
+{
+    \"resourceMetrics\": [
+      {
+        \"resource\": {
+          \"attributes\": [
+            {
+              \"key\": \"service.name\",
+              \"value\": {
+                \"stringValue\": \"${URL}\"
+              }
+            }
+          ]
+        },
+        \"scopeMetrics\": [
+          {
+            \"metrics\": [
+              {
+                \"name\": \"tls.server.not_after.time_left\",
+                \"unit\": \"s\",
+                \"description\": \"\",
+                \"gauge\": {
+                  \"dataPoints\": [
+                    {
+                      \"asInt\": ${secondsLeft},
+                      \"timeUnixNano\": ${now}000000000
+                    }
+                  ]
+                }
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+"
+curl -X POST -H "Content-Type: application/json" -d "${data}" -i localhost:4318/v1/metrics
+```
+
+Activate the OTLP Receiver in your `collectors-config`:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      http:
+      grpc:
+exporters:
+  debug:
+    verbosity: detailed
+service:
+  pipelines:
+    metrics:
+      receivers: [otlp]
+      exporters: [debug]
+```
+
+Execute your updated `ssl_check.sh`:
+
+```shell
+./ssl_check.sh opentelemetry.io
+```
+
+This time, your metric will be presented with the correct unit set and the
+resource reported as defined in your JSON:
+
+```shell
```

Comment:
Hmm, this isn't `shell` code, so maybe leave blank or find another better lang?

~~~suggestion
```
~~~